post-policy.go

```diff
@@ -302,11 +302,31 @@ func (p *PostPolicy) SetUserMetadata(key, value string) error {
 }
 
 // SetChecksum sets the checksum of the request.
-func (p *PostPolicy) SetChecksum(c Checksum) {
+func (p *PostPolicy) SetChecksum(c Checksum) error {
 	if c.IsSet() {
 		p.formData[amzChecksumAlgo] = c.Type.String()
 		p.formData[c.Type.Key()] = c.Encoded()
+
+		// Needed for S3 compatibility. MinIO ignores the checksum keys in the policy.
+		// https://github.com/minio/minio/blob/RELEASE.2024-08-29T01-40-52Z/cmd/postpolicyform.go#L60-L65
```

Comment:
@jiuker You added this in https://github.com/minio/minio/pull/18074

I don't understand why these should be ignored, which seems like a bug.

Reply:
I spoke to Jiuker and he agrees it is a bug, so I will remove the ignores in a PR to minio/minio